src/modules/tools/zprobe/CalibrationStrategy.h

```diff
@@ -0,0 +1,64 @@
+// Generic calibration strategy based on nonlinear optimization
+//
+// Copyright (c) Oskar Linde 2015
+//
+// Licence: GPL v2
+
+#ifndef _GENERICLEVELINGSTRATEGY
+#define _GENERICLEVELINGSTRATEGY
+
+#include "LevelingStrategy.h"
+#include <string>
+#include <vector>
+
+#define calibration_strategy_checksum CHECKSUM("calibration")
+struct V3 { float m[3]; };
+
+class CalibrationStrategy : public LevelingStrategy
+{
+public:
+    CalibrationStrategy(class ZProbe *zprobe) : LevelingStrategy(zprobe){}
+    ~CalibrationStrategy(){}
+    bool handleGcode(Gcode* gcode);
+    bool handleConfig();
+
+    // probes n points, return actuator positions of trigger points
+    bool probe_pattern(int n, int repeats, float actuator_positions[/*N*/][3]);
+
+private:
+    // Radius of probe circle. TODO: model other build plate shapes
+    float probe_radius;
+
+    // Plane tilt
+    float plane_u = 0, plane_v = 0;
```

Comment:
Actually, both syntaxes are allowed. The standard says: "if the entity is a non-static data member that has a brace-or-equal-initializer, the entity is initialized as specified in 8.5; (brace-or-equal-initializer). Anyway, happy to change if it makes it more consistent with other parts of the code.